pkg/R/context.R

```diff
@@ -85,11 +85,9 @@ parallelize <- function(sc, coll, numSlices = 1) {
   # TODO: bound/safeguard numSlices
   # TODO: unit tests for if the split works for all primitives
   # TODO: support matrix, data frame, etc
-  if (!is.list(coll)) {
-    if (!is.vector(coll)) {
-      message(paste("context.R: parallelize() currently only supports lists and vectors.",
-                    "Calling as.list() to coerce coll into a list."))
-    }
+  if ((!is.list(coll) && !is.vector(coll)) || is.data.frame(col)) {
```

Comment:
Sorry, let me fix and resubmit.

 

 

 

From: Shivaram Venkataraman [mailto:notifications@github.com] 
Sent: Sunday, February 01, 2015 9:58 PM
To: amplab-extras/SparkR-pkg
Cc: dputler
Subject: Re: [SparkR-pkg] Have data frames handled appropriately by parallize (#150)

 

In pkg/R/context.R <https://github.com/amplab-extras/SparkR-pkg/pull/150#discussion_r23908070> :

> @@ -85,11 +85,9 @@ parallelize <- function(sc, coll, numSlices = 1) {
>    # TODO: bound/safeguard numSlices
>    # TODO: unit tests for if the split works for all primitives
>    # TODO: support matrix, data frame, etc
> -  if (!is.list(coll)) {
> -    if (!is.vector(coll)) {
> -      message(paste("context.R: parallelize() currently only supports lists and vectors.",
> -                    "Calling as.list() to coerce coll into a list."))
> -    }
> +  if ((!is.list(coll) && !is.vector(coll)) || is.data.frame(col)) {

This should be is.data.frame(coll) (i.e. the variable has two ls)

—
Reply to this email directly or view it on GitHub <https://github.com/amplab-extras/SparkR-pkg/pull/150/files#r23908070> .  <https://github.com/notifications/beacon/AIGvMg2qlEW-YLsBwG8-PJQvVfbeZ1Tbks5nnwlogaJpZM4DaK_g.gif>